dot/parachain/availability-store/availabilitystore.go

```diff
@@ -1,33 +1,159 @@
+	"github.com/ChainSafe/gossamer/lib/common"
 )
 
 var logger = log.NewFromGlobal(log.AddContext("pkg", "parachain-availability-store"))
 
+const (
+	avaliableDataPrefix = "available"
+	chunkPrefix         = "chunk"
+	metaPrefix          = "meta"
+	unfinalizedPrefix   = "unfinalized"
+	pruneByTimePrefix   = "prune_by_time"
+)
+
+// AvailabilityStoreSubsystem is the struct that holds subsystem data for the availability store
 type AvailabilityStoreSubsystem struct {
 	SubSystemToOverseer chan<- any
 	OverseerToSubSystem <-chan any
-	//db            interface{} // Define the actual database type
-	//config        Config      // Define the actual config type
-	//pruningConfig PruningConfig
-	//clock         Clock
-	//metrics       Metrics
+	availabilityStore   AvailabilityStore
+	//TODO: pruningConfig PruningConfig
+	//TODO: clock         Clock
+	//TODO: metrics       Metrics
+}
+
+// AvailabilityStore is the struct that holds data for the availability store
+type AvailabilityStore struct {
+	availableTable database.Table
+	chunkTable     database.Table
+	metaTable      database.Table
+	//TODO: unfinalizedTable database.Table
+	//TODO: pruneByTimeTable database.Table
+}
+
+// NewAvailabilityStore creates a new instance of AvailabilityStore
+func NewAvailabilityStore(db database.Database) (*AvailabilityStore, error) {
+	return &AvailabilityStore{
+		availableTable: database.NewTable(db, avaliableDataPrefix),
+		chunkTable:     database.NewTable(db, chunkPrefix),
+		metaTable:      database.NewTable(db, metaPrefix),
+	}, nil
+}
+
+// loadAvailableData loads available data from the availability store
+func (as *AvailabilityStore) loadAvailableData(candidate common.Hash) (AvailableData, error) {
+	resultBytes, err := as.availableTable.Get(candidate[:])
+	if err != nil {
+		return AvailableData{}, err
+	}
+	result := AvailableData{}
+	err = json.Unmarshal(resultBytes, &result)
+	return result, err
+}
+
+// loadMetaData loads metadata from the availability store
+func (as *AvailabilityStore) loadMetaData(candidate common.Hash) (CandidateMeta, error) {
+	resultBytes, err := as.metaTable.Get(candidate[:])
+	if err != nil {
+		return CandidateMeta{}, err
+	}
+	result := CandidateMeta{}
+	err = json.Unmarshal(resultBytes, &result)
+	return result, err
+}
+
+// storeMetaData stores metadata in the availability store
+func (as *AvailabilityStore) storeMetaData(candidate common.Hash, meta CandidateMeta) error {
+	dataBytes, err := json.Marshal(meta)
+	if err != nil {
+		return err
+	}
+	return as.metaTable.Put(candidate[:], dataBytes)
+}
+
+// loadChunk loads a chunk from the availability store
+func (as *AvailabilityStore) loadChunk(candidate common.Hash, validatorIndex uint32) (ErasureChunk, error) {
+	resultBytes, err := as.chunkTable.Get(append(candidate[:], uint32ToBytes(validatorIndex)...))
+	if err != nil {
+		return ErasureChunk{}, err
+	}
+	result := ErasureChunk{}
+	err = json.Unmarshal(resultBytes, &result)
+	return result, err
+}
```

Comment:
- wrap errors 
- written pointer

Looks like this is happening at many places. I will avoid commenting everywhere.